### PR TITLE
refactor(app): Added solana send txn support

### PR DESCRIPTION
**Pull request by vnksnkr**
Added support for Solana Send Transaction Flow

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4965 Solana send transaction](https://app.clickup.com/t/9002019994/PRF-4965)

File: apps/solana_app/solana_sign_txn.c [hunk 1]
```diff
@@ -448,9 +451,7 @@ static bool send_signature(solana_query_t *query,
          sizeof(solana_sign_txn_signature_response_t));
 
   solana_send_result(&result);
```
Comment (ujjwal-cyph) [inline apps/solana_app/solana_sign_txn.c:453]: can we split this to sign & send functions separately? that way, the secret will be inside one function unit only. Currently, it's split over fetch_seed and send_signature. refer: https://github.com/Cypherock/x1_wallet_firmware/blob/refactor/core-framework/apps/near_app/near_txn.c#L387-L388

**Review by ujjwal-cyph — APPROVED** [review]
LGTM

File: apps/solana_app/solana_sign_txn.c (hunk 1 above)
Comment (amanCypherock) [reply apps/solana_app/solana_sign_txn.c:453]: Will look into it

Comment (amanCypherock) [reply apps/solana_app/solana_sign_txn.c:453]: Will take it up as a follow up change.

Comment (ujjwal-cyph) [reply apps/solana_app/solana_sign_txn.c:453]: okay.

**Pull request merged**